From 310d7d83738475b0ce593ea6cb73b796a6a0327f Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Thu, 20 Jun 2019 01:12:55 +0200
Subject: [PATCH 28/36] SystemUI: Set MobileGroup view visibility
 unconditionally

* We need to do it so that MobileGroup view visibility
  is initialized properly, otherwise it gets shown if
  you disable one of SIM cards and restart SystemUI,
  change visibility of any status bar icon, toggle
  airplane mode on and off or simply reboot your
  device (assuming the device is using real QCOM
  qti-telephony-common implementation).
* By the way setVisibility() is smart enough to not
  do any 'hard work' when the flags are unchanged so
  we shouldn't see any perf regressions or anything
  like this.
* Fixes:
  - https://gitlab.com/LineageOS/issues/android/issues/294
  - https://gitlab.com/LineageOS/issues/android/issues/437

Change-Id: I7f620836a53b90426fd75d8c1e8b0f6bad6688dd
---
 .../android/systemui/statusbar/StatusBarMobileView.java   | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarMobileView.java b/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarMobileView.java
index 0f65cab879f..45bf8268d2b 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarMobileView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarMobileView.java
@@ -145,7 +145,7 @@ public class StatusBarMobileView extends FrameLayout implements DarkReceiver,
 
     private void initViewState() {
         setContentDescription(mState.contentDescription);
-        if (!mState.visible) {
+        if (!mState.visible || !mState.provisioned) {
             mMobileGroup.setVisibility(View.GONE);
         } else {
             mMobileGroup.setVisibility(View.VISIBLE);
@@ -169,10 +169,8 @@ public class StatusBarMobileView extends FrameLayout implements DarkReceiver,
 
     private void updateState(MobileIconState state) {
         setContentDescription(state.contentDescription);
-        if (mState.visible != state.visible || mState.provisioned != state.provisioned) {
-            mMobileGroup.setVisibility(state.visible && state.provisioned
-                    ? View.VISIBLE : View.GONE);
-        }
+        mMobileGroup.setVisibility(state.visible && state.provisioned
+                ? View.VISIBLE : View.GONE);
         if (mState.strengthId != state.strengthId) {
             mMobileDrawable.setLevel(state.strengthId);
         }
-- 
2.17.1

